### PR TITLE
Fix #15093, also displays alloy names before combining materials in the nanocrucible.

### LINKED ISSUE #15093: duplicate numbers when using the nanocrucible
### Labels

[BUG]

### Description

theres a duplicate number in the text "You make 1 1 Stpper bar.". This happens for any combo.

### Steps to reproduce

1.  Put 2 materials in the nanocrucible
2. Combine them.
3. See error.

### Expected Behavior

I expect to see "You make 1 Stpper bar."

### Additional Information & Screenshots

_No response_

### PR DESCRIPTION
**Pull request by MRLinew**
## About the PR 

This pr is just 2 simple fixes for minor issues with the nanocrucible. It fixes duplicate numbers that have been around for awhile (#15093) and it displays alloy names before you actually press "combine"

## Why's this needed? 
With censorium being a thing, theres no need for "cock" to still display in the nanocrucible when making it. Also, its a nice feature being able to confirm an alloy without wasting materials on a mixup.

## Changelog 

+ The Nano-crucible now displays alloy names before you fabricate them.

### CODE REVIEW

File: +secret
Comment (MRLinew) [inline +secret:1]: im not sure if this does anything or how this is here~

**Comment by MylieDaniels** [comment]
Would it be possible to instead find a way to find alloy names beforehand?  That would be a nice feature AND fix this issue (and any further cases of it).

File: +secret
Comment (TobleroneSwordfish) [reply +secret:1]: well this is the "file" that determines what secret module commit gets used, no idea how you managed to commit it but merging from upstream master might fix it?

Comment (MRLinew) [reply +secret:1]: yeah i still need help with this. The codes the easy part the github isnt. How to i untouch the secret module (not that i touched it anyway 😭

**Comment by MRLinew** [comment]
> Would it be possible to instead find a way to find alloy names beforehand? That would be a nice feature AND fix this issue (and any further cases of it).

done

File: code/modules/materials/Mat_ProcsDefines.dm [hunk 1]
```diff
@@ -352,6 +352,12 @@ var/global/list/material_cache = buildMaterialCache()
 		if(R.validate(M)) return R
 	return null
 
+/proc/findRecipeName(var/obj/item/One,var/obj/item/Two)
+	var/tempmerge = getFusedMaterial(One.material, Two.material)
+	for(var/datum/material_recipe/R in materialRecipes)
+		if(R.validate(tempmerge)) return R
```
Comment (TobleroneSwordfish) [inline code/modules/materials/Mat_ProcsDefines.dm:358]: shouldn't this be `return R.getName()`?

Comment (MRLinew) [reply code/modules/materials/Mat_ProcsDefines.dm:358]: yeah probably.

Comment (MRLinew) [reply code/modules/materials/Mat_ProcsDefines.dm:358]: it doesnt work with R.getName() but it works the original way

**Pull request merged**